snapcraft/plugins/v2/ruby.py

```diff
@@ -0,0 +1,156 @@
+dependencies from a `Gemfile` found in the sources.
+
+Additionally, this plugin uses the following plugin-specific keywords:
+    - ruby-version:
+      (string)
+      The version of ruby you want this snap to run. (e.g. '3.0' or '2.7.2')
+    - ruby-flavor:
+      (string)
+      Other flavors of ruby supported by ruby-install (e.g. 'jruby', ...)
+    - gems:
+      (list)
+      A list of gems to install.
+    - use-bundler
+      (boolean)
+      Use bundler to install gems from a Gemfile (defaults 'false').
+    - use-jemalloc:
+      (boolean)
+      Build ruby with libjemalloc (defaults 'false').
+    - shared:
+      (boolean)
+      Build ruby as a shared library (defaults 'false').
+    - configure-options:
+      (array of strings)
+      Additional configure options to use when configuring ruby.
+"""
+
+import os
+import re
+import logging
+
+from typing import Any, Dict, List, Set
+from snapcraft.plugins.v2 import PluginV2
+
+
+class RubyPlugin(PluginV2):
+    @classmethod
+    def get_schema(cls) -> Dict[str, Any]:
+        return {
+            "$schema": "http://json-schema.org/draft-04/schema#",
+            "type": "object",
+            "additionalProperties": False,
+            "properties": {
+                "ruby-flavor": {
+                    "type": "string",
+                    "default": "ruby",
+                },
+                "ruby-version": {
+                    "type": "string",
+                    "default": "3.0",
+                    "pattern": r"^\d+\.\d+(\.\d+)?$",
+                },
+                "use-bundler": {
+                    "type": "boolean",
+                    "default": False,
+                },
+                "use-jemalloc": {
+                    "type": "boolean",
+                    "default": False,
+                },
+                "shared": {
+                    "type": "boolean",
+                    "default": False,
+                },
+                "configure-options": {
+                    "type": "array",
+                    "items": {"type": "string"},
+                    "default": [],
+                },
+                "gems": {
+                    "type": "array",
+                    "items": {"type": "string"},
+                    "default": [],
+                },
+            },
+        }
+
+    def get_build_snaps(self) -> Set[str]:
+        return set()
+
+    def get_build_packages(self) -> Set[str]:
+        packages = {"curl", "jq"}
+
+        if self.options.use_jemalloc:
+            packages.add("libjemalloc-dev")
+
+        return packages
+
+    def get_build_environment(self) -> Dict[str, str]:
+        env = {
+            "PATH": "${SNAPCRAFT_PART_INSTALL}/bin:${PATH}",
+        }
+
+        if self.options.shared:
+            # for finding ruby.so when running `gem` or `bundle`
+            env["LD_LIBRARY_PATH"] = "${SNAPCRAFT_PART_INSTALL}/lib:${LD_LIBRARY_PATH}"
```

Comment:
To expand correctly if LD_LIBRARY_PATH is empty:
```suggestion
            env["LD_LIBRARY_PATH"] = "${SNAPCRAFT_PART_INSTALL}/lib:${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}"
```

Reply:
👍